File: app/src/main/java/com/android/kotemanagement/room/repository/SoldiersRepository.java
package com.android.kotemanagement.room.repository;

import android.app.Application;

import androidx.lifecycle.LiveData;

import com.android.kotemanagement.room.dao.SoldiersDao;
import com.android.kotemanagement.room.database.KoteDatabase;
import com.android.kotemanagement.room.entities.Soldiers;

import java.util.List;
import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

public class SoldiersRepository {
    private SoldiersDao soldiersDao;

    private LiveData<List<Soldiers>> getAllSoldiersList;
    private KoteDatabase koteDatabase;

    private Executor executor;

    public SoldiersRepository(Application application) {
        executor = Executors.newSingleThreadExecutor();
        koteDatabase = KoteDatabase.getDatabaseInstance(application);
        soldiersDao = koteDatabase.getSoldiersDao();
        getAllSoldiersList = soldiersDao.getAllSoldiersList();

    }

    public void insert(Soldiers soldiers) {
        executor.execute(() -> soldiersDao.insert(soldiers));
    }

    public void update(Soldiers soldiers) {
        executor.execute(() -> soldiersDao.update(soldiers));
    }

    public void delete(Soldiers soldiers) {
        executor.execute(() -> soldiersDao.delete(soldiers));
    }

    public Soldiers getSoldierByArmyNumber(String armyNumber) {
        return soldiersDao.getSoldierByArmyNumber(armyNumber);
    }

    public LiveData<List<Soldiers>> getAllSoldiersList() {
        return getAllSoldiersList;
    }

}

File: app/src/main/java/com/android/kotemanagement/roomdatabase/repository/SoldiersRepository.kt
package com.android.kotemanagement.roomdatabase.repository

import android.app.Application
import androidx.lifecycle.LiveData
import com.android.kotemanagement.roomdatabase.dao.SoldiersDAO
import com.android.kotemanagement.roomdatabase.database.KoteDatabase
import com.android.kotemanagement.roomdatabase.entities.Soldiers

class SoldiersRepository(application: Application) {
    var soldiersDAO : SoldiersDAO
    var allSoldiersList : LiveData<List<Soldiers>>

    init {
        val database = KoteDatabase.getDatabase(application)
        soldiersDAO = database.getSoldiersDao()
        allSoldiersList = soldiersDAO.getAllSoldiers()
    }

    suspend fun insert(soldier: Soldiers) {
        soldiersDAO.insert(soldier)
    }

    suspend fun delete(soldier: Soldiers) {
        soldiersDAO.delete(soldier)
    }

    suspend fun update(soldier: Soldiers) {
        soldiersDAO.delete(soldier)
    }

    suspend fun getSoldierByArmyNumber(armyNumber: String) : Soldiers {
        return soldiersDAO.getSoldierByArmyNumber(armyNumber)
    }

    fun getAllSoldiers() : LiveData<List<Soldiers>> {
        return allSoldiersList
    }

}